### PR TITLE
Models docstring

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR cleans up all docstrings following up from #20757 and #21199. It removes the need for the `processor_class` in TensorFlow and Flax generic examples by setting in the examples like #20757 did for PyTorch then makes a full pass across all models to clean up the docstrings (removing the processor_class` in the `add_code_sample` decorator, remove random outputs, use the auto classes for preprocessing).

Note that in some cases we can't use the auto-classes for preprocessing: when linking to the `__call__` method of a processor or image processor, we need the actual class (cc @amyeroberts I changed a couple of things you did here).

### CODE REVIEW

**Comment by ydshieh** [comment]
Thank you @sgugger for cleaning this up. With all ~250 files, I will try you instead of look lines by lines, except one question below.

I would definitely prefer to run a doctest first offline before merging this PR - for which I can launch on my side. From previous PRs, it has shown there are always some surprise.

> Note that in some cases we can't use the auto-classes for preprocessing: when linking to the __call__ method of a processor or image processor, we need the actual class (cc @amyeroberts I changed a couple of things you did here).

I see even in such places, we still have
```python
Pixel values can be obtained using [`AutoImageProcessor`].  See [`ConvNextImageProcessor.__call__`] for details.
```
I don't have much context and prior knowledge, but is it true we want to use `AutoImageProcessor` but `ConvNextImageProcessor.__call__` in such cases?

**Comment by sgugger** [comment]
> With all ~250 files, I will trust you instead of look lines by lines.

A review would still be much appreciated, as it could catch accidental typos.

> I would definitely prefer to run a doctest first offline before merging this PR - for which I can launch on my side. From previous PRs, it has shown there are always some surprise. I will launch doctest CI when all reviewers give their approval.

Sure, we can wait for that as long as the results are available before the release branch is cut.

> I don't have much context and prior knowledge, but is it true we want to use AutoImageProcessor but ConvNextImageProcessor.__call__ in such cases?

Yes.

**Comment by ydshieh** [comment]
I triggered the doctest CI against the (last) commit (so far) in this PR. Will take a look on the PR changes too :-)

File: src/transformers/models/bert/modeling_tf_bert.py [hunk 1]
```diff
@@ -80,8 +79,6 @@
 _CHECKPOINT_FOR_QA = "ydshieh/bert-base-cased-squad2"
 _QA_EXPECTED_OUTPUT = "'a nice puppet'"
 _QA_EXPECTED_LOSS = 7.41
-_QA_TARGET_START_INDEX = 14
```
Comment (ydshieh) [inline src/transformers/models/bert/modeling_tf_bert.py:83]: Currently these are not used, so I understand the reason you remove them.

But we should instead keep here and use them in `add_code_sample_docstrings` for `BertForQuestionAnswering`.

(without using it, it works currently due to the default values `14` and `15` specified in `def add_code_sample_docstrings` - this is not the best approach though)

File: src/transformers/models/blenderbot/modeling_tf_blenderbot.py [hunk 2]
```diff
@@ -536,12 +535,11 @@ def serving(self, inputs):
...
-        >>> from transformers import BlenderbotTokenizer, TFBlenderbotForConditionalGeneration >>> mname =
+        >>> from transformers import AutoTokenizer, TFBlenderbotForConditionalGeneration >>> mname =
         'facebook/blenderbot-400M-distill' >>> model = TFBlenderbotForConditionalGeneration.from_pretrained(mname) >>>
-        tokenizer = BlenderbotTokenizer.from_pretrained(mname) >>> UTTERANCE = "My friends are cool but they eat too
-        many carbs." >>> print("Human: ", UTTERANCE) >>> inputs = tokenizer([UTTERANCE], return_tensors='tf') >>>
-        reply_ids = model.generate(**inputs) >>> print("Bot: ", tokenizer.batch_decode(reply_ids,
-        skip_special_tokens=True)[0])
+        tokenizer = AutoTokenizer.from_pretrained(mname) >>> UTTERANCE = "My friends are cool but they eat too many
+        carbs." >>> print("Human: ", UTTERANCE) >>> inputs = tokenizer([UTTERANCE], return_tensors='tf') >>> reply_ids
+        = model.generate(**inputs) >>> print("Bot: ", tokenizer.batch_decode(reply_ids, skip_special_tokens=True)[0])
```
Comment (LysandreJik) [inline src/transformers/models/blenderbot/modeling_tf_blenderbot.py:543]: There's an issue in this docstring's format

File: src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py [hunk 3]
```diff
@@ -1497,9 +1494,9 @@ def update_inputs_for_generation(self, model_outputs, model_kwargs):
 
     Mask filling example:
 
-        >>> from transformers import BlenderbotSmallTokenizer, FlaxBlenderbotSmallForConditionalGeneration >>>
-        tokenizer = BlenderbotSmallTokenizer.from_pretrained('facebook/blenderbot_small-90M') >>> TXT = "My friends are
-        <mask> but they eat too many carbs."
+        >>> from transformers import AutoTokenizer, FlaxBlenderbotSmallForConditionalGeneration >>> tokenizer =
+        AutoTokenizer.from_pretrained('facebook/blenderbot_small-90M') >>> TXT = "My friends are <mask> but they eat
+        too many carbs."
```
Comment (LysandreJik) [inline src/transformers/models/blenderbot_small/modeling_flax_blenderbot_small.py:1499]: same here

File: src/transformers/models/blenderbot_small/modeling_tf_blenderbot_small.py [hunk 4]
```diff
@@ -536,9 +535,9 @@ def serving(self, inputs):
 BLENDERBOT_SMALL_GENERATION_EXAMPLE = r"""
     Conversation example::
 
-        >>> from transformers import BlenderbotSmallTokenizer, TFBlenderbotSmallForConditionalGeneration >>> mname =
+        >>> from transformers import AutoTokenizer, TFBlenderbotSmallForConditionalGeneration >>> mname =
         'facebook/blenderbot_small-90M' >>> model = BlenderbotSmallForConditionalGeneration.from_pretrained(mname) >>>
-        tokenizer = BlenderbotSmallTokenizer.from_pretrained(mname)
+        tokenizer = AutoTokenizer.from_pretrained(mname)
```
Comment (LysandreJik) [inline src/transformers/models/blenderbot_small/modeling_tf_blenderbot_small.py:540]: here too

**Review by LysandreJik — APPROVED** [review]
Impressive PR! Only found a couple of typos.

File: src/transformers/models/tapas/modeling_tapas.py [hunk 5]
```diff
@@ -43,7 +43,6 @@
 logger = logging.get_logger(__name__)
 
 _CONFIG_FOR_DOC = "TapasConfig"
-_TOKENIZER_FOR_DOC = "TapasTokenizer"
 _TOKENIZER_FOR_DOC = "google/tapas-base"
```
Comment (ydshieh) [inline src/transformers/models/tapas/modeling_tapas.py:46]: `_TOKENIZER_FOR_DOC = "google/tapas-base"`  one line below should be deleted too.

**Review by ydshieh — APPROVED** [review]
Leave some minor comments. (one is mentioned by Lysandre already I believe).
Thank you again!

File: src/transformers/models/gpt2/modeling_gpt2.py [hunk 6]
```diff
@@ -1343,12 +1340,9 @@ def __init__(self, config):
 
     @add_start_docstrings_to_model_forward(GPT2_INPUTS_DOCSTRING)
     @add_code_sample_docstrings(
-        processor_class=_TOKENIZER_FOR_DOC,
         checkpoint="microsoft/DialogRPT-updown",
         output_type=SequenceClassifierOutputWithPast,
         config_class=_CONFIG_FOR_DOC,
-        expected_output="'LABEL_0'",
-        expected_loss=5.28,
```
[the remark this answers is not in the document]
Comment (sgugger) [reply src/transformers/models/gpt2/modeling_gpt2.py:1351]: LABEL_0 is not informative at all and the loss seems very high, I don't think this is a model trained for sequence classification.

File: src/transformers/models/blenderbot/modeling_blenderbot.py [hunk 7]
```diff
@@ -50,8 +50,6 @@
 logger = logging.get_logger(__name__)
 
 _CONFIG_FOR_DOC = "BlenderbotConfig"
-_TOKENIZER_FOR_DOC = "BlenderbotTokenizer"
-_CHECKPOINT_FOR_DOC = "facebook/blenderbot-400M-distill"
```
[the remark this answers is not in the document]
Comment (sgugger) [reply src/transformers/models/blenderbot/modeling_blenderbot.py:54]: Removing the `_CHECKPOINT_FOR_DOC` here is my mistake.
